File: src/gig/router.ts
import { Router } from 'express';

import {
  handleGetGigs,
  handleCreateGig,
  handleGetGig,
  handleUpdateGig,
  handleDeleteGig,
} from './controller';
import authMiddleware from '../auth/middleware';

const gigRouter = Router();

gigRouter.get('/', handleGetGigs);
gigRouter.get('/:id', handleGetGig);
gigRouter.post('/', authMiddleware, handleCreateGig);
gigRouter.put('/:id', authMiddleware, handleUpdateGig);
gigRouter.delete('/:id', authMiddleware, handleDeleteGig);

export default gigRouter;

File: gigs/router.go
package gigs

import (
	"github.com/AndreyGalchevski/strident-api/auth"
	"github.com/gorilla/mux"
)

func InitGigsRouter(r *mux.Router) {
	r.HandleFunc("/gigs", handleGetGigs).Methods("GET")

	r.HandleFunc("/gigs/{id}", auth.VerifyAuthorization(handleGetGigByID)).Methods("GET")
	r.HandleFunc("/gigs", auth.VerifyAuthorization(handlePostGig)).Methods("POST")
	r.HandleFunc("/gigs/{id}", auth.VerifyAuthorization(handlePatchGig)).Methods("PATCH")
	r.HandleFunc("/gigs/{id}", auth.VerifyAuthorization(handleDeleteGig)).Methods("DELETE")
}
